contributing.md

```diff
@@ -0,0 +1,171 @@
+
+The CLI comes built with a debug option (`-d/--debug`) which will include a `debugOutput` property on the results object containing any CQL, ELM, ValueSets, and engine results processed during execution. These files will also be written to a `./debug` directory when this flag is provided,
+which is useful for inspecting contents of ELM, CQL, or results.
+
+# Creating Measure Bundles for Specific Cases
+
+Most commonly, development on `fqm-execution` will involve pre-existing measure bundles provided via a content repository or some other means (e.g. attached to a GitHub issue report). However, in some cases it may be useful to create a smaller "stripped down" version of a measure bundle
+that makes it easier to debug potential issues.
+
+To create your own measure bundle to use during development, consider using the [ecqm-bundler command line utility](https://github.com/mgramigna/ecqm-bundler) to easily generate valid measure bundles that `fqm-execution` will accept as input. Alongside this, you can also use our [publicly deployed instance](https://projecttacoma.github.io/fqm-testify/) of [fqm-testify](https://github.com/projecttacoma/fqm-testify)
+to create simple patients that you can run through `fqm-execution` as well.
+
+# Debugging
+
+## Common Issues
+
+In general, runtime errors that come directly from `fqm-execution` are quite rare. Most unhandled errors come from dependent libraries, i.e. [cql-execution](https://github.com/cqframework/cql-execution) or [cql-exec-fhir](https://github.com/cqframework/cql-exec-fhir)
+
+### `x.someFunction` is Not a Function
+
+Error throws with this message commonly come from `cql-execution`. The most common cause for this kind of error is a mismatch in data types for either the ELM expressions or the patient data itself (e.g. trying to do a `union` on arguments that are not list-like). The recommended approach for debugging this kind of issue is:
+
+1. Try to isolate the ELM library and expression where the issue comes from
+2. Visually inspect the content of the ELM JSON for this library, and ensure that the operands for the function being called are the correct data types (e.g. see if it should be wrapped in a `FHIRHelpers.To___` function call)
+3. Ensure that the FHIR data present in the patient `Bundle` are valid according to the FHIR specification
+
+If the above conditions do not lead to an identified cause, it might be worth raising an issue in the [cql-execution](https://github.com/cqframework/cql-execution) repository. To verify that the issue indeed comes from `cql-execution`, consider using the [execution extractor command line utility](https://github.com/mgramigna/execution-extractor) to extract all of the ELM content, ValueSets, etc. and run it through `cql-execution` directly without needing to go through `fqm-execution`. If the error still persists, then the issue lies outside `fqm-execution`, and it either a problem with the input or with `cql-execution` itself.
```

Comment:
```suggestion
If the above conditions do not lead to an identified cause, it might be worth raising an issue in the [cql-execution](https://github.com/cqframework/cql-execution) repository. To verify that the issue indeed comes from `cql-execution`, consider using the [execution extractor command line utility](https://github.com/mgramigna/execution-extractor) to extract all of the ELM content, ValueSets, etc. and run it through `cql-execution` directly without needing to go through `fqm-execution`. If the error still persists, then the issue lies outside `fqm-execution`, and is either a problem with the input or with `cql-execution` itself.
```